include/zephyr/drivers/spi.h

```diff
@@ -1004,7 +1004,9 @@ static inline int spi_read_dt(const struct spi_dt_spec *spec,
  * @param tx_bufs Buffer array where data to be sent originates from.
  *
  * @retval 0 If successful.
- * @retval -ENOTSUP or -EINVAL means some part of the spi config is not possible for device
+ * @retval -ENOTSUP means some part of the spi config is not supported either by the
+ *	   device hardware or the driver software.
+ * @retval -EINVAL means that some parameter of the spi_config is invalid for the device.
```

Comment:
perhaps "for the device" is superfluous. It's not device specific here. It's like giving a wrong combination of config that would just not work anywhere.